### PR TITLE
Give a valid service address in SRV lookup when it differs from the node's

### LINKED ISSUE #832: SRV records that point to names that don't resolve to the service's IP
I'm having an issue where:

* Software is using a SRV record via consul to find a service, which returns the appropriate port for the service, and the FQDN for the relevant agent's node. The additional info section includes a record for that name, mapping to the service's IP.
* The software is disregarding the additional info and doing a lookup for the FQDN itself, which resolves to a totally different IP, which is not the one the service is listening on.

It seems reasonable to me that SRV records could respond with an FQDN that leads to the service itself, although I'm not sure how that would look.

I'm also unsure if the behavior of this software (srv-router in this case) is typical, but it would not surprise me if this came up in other instances.

### PR DESCRIPTION
**Pull request by kyhavlov**
Fixes #832 

Still need to add a 'TestDNS_ServiceLookup_ServiceAddressIPv6' test, I think that's the only thing missing here.

### CODE REVIEW

File: command/agent/dns.go [hunk 1]
```diff
@@ -357,6 +358,46 @@ PARSE:
 		query := strings.Join(labels[:n-1], ".")
 		d.preparedQueryLookup(network, datacenter, query, req, resp)
 
+	case "addr":
+		if n < 2 {
```
Comment (slackpad) [inline command/agent/dns.go]: This should probably be `!= 2` so we don't accept any periods, since we know it's just hex stuff.

**Comment by slackpad** [comment]
LGTM - one tiny thing and then the IPv6 test and this should be gtg!

File: command/agent/dns_test.go [hunk 2]
```diff
@@ -1062,6 +1062,100 @@ func TestDNS_ServiceLookup_ServiceAddress(t *testing.T) {
...
+		}
+		if srvRec.Hdr.Ttl != 0 {
+			t.Fatalf("Bad: %#v", in.Answer[0])
+		}
+
+		aRec, ok := in.Extra[0].(*dns.AAAA)
+		if !ok {
+			t.Fatalf("Bad: %#v", in.Extra[0])
+		}
+		if aRec.Hdr.Name != "2607002040050808000000000000200e.addr.dc1.consul." {
```
Comment (slackpad) [inline command/agent/dns_test.go]: lol these hex names are 1337

**Comment by kyhavlov** [comment]
Addressed the comment and added a couple tests for lookups on 'addr.consul', since the existing ones didn't actually do that

**Pull request merged**